config/jobs/periodic/docker-in-docker/periodic-build-docker.yaml

```diff
@@ -0,0 +1,52 @@
+periodics:
+  - name: periodic-build-docker
+    cluster: k8s-ppc64le-cluster
+    decorate: true
+    reporter_config:
+      slack:
+        channel: 'prow-job-notifications'
+        job_states_to_report:
+         - success
+        report_template: 'Job {{.Spec.Job}} of type {{.Spec.Type}} ended with state {{.Status.State}}. <!subteam^S01E9J0G3EU> <{{.Status.URL}}|View logs>'
+    decoration_config:
+      gcs_configuration:
+        bucket: s3://ppc64le-docker
+        path_strategy: explicit
+      gcs_credentials_secret: s3-auth
+      timeout: 20h
+    extra_refs:
+    - org: ppc64le-cloud
+      repo: docker-ce-build
+      base_ref: main
+      workdir: false
+    interval: 120h
+    spec:
+      containers:
+      - image: quay.io/powercloud/docker-ce-build
+        command:
+        - /bin/bash
+        args:
+        - -c
+        - |
+          if [[ $PWD != /home/prow/go/src/github.com/ppc64le-cloud/docker-ce-build ]]
+          then
+            cd /home/prow/go/src/github.com/ppc64le-cloud/docker-ce-build
+           fi
+           chmod a+x *.sh
+          source prowjob-periodic-dind-build.sh
+        resources: {}
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: docker-root
+          mountPath: /var/lib/docker
+        env:
+          - name: S3_SECRET_AUTH
+            valueFrom:
+              secretKeyRef:
+                name: secret-s3
+                key: password
+      restartPolicy: Never
```

Comment:
Building the packages and testing them takes a long time. That is why I had changed the timeout to 18 hours instead of 2 hours. Restarting it when it fails, did not seem efficient in my opinion, since the timeout would eventually stop the pod before being completed.

Reply:
Isn't a timeout of 20 hours too long for a job?
What is the average time that the job takes to build the packages?

Also, this pod has the pod-utility containers like cloneref, initupload, etc that are helpers for the actual job.
IMO, due to connectivity issues or any other reason, they might be certain failures that can be fixed by a simple restart. Leaving the default policy of `Always` shall help in such cases.

Reply:
Building the packages takes a long time. It takes at least 5 hours. It can take up 12 hours. And the testing part takes 1 hour per distribution.